lib/fog/compute/google/models/instance_template.rb

```diff
@@ -0,0 +1,33 @@
+module Fog
+  module Compute
+    class Google
+      class InstanceTemplate < Fog::Model
+        identity :name
+
+        attribute :kind
+        attribute :self_link, :aliases => "selfLink"
+        attribute :description
+        # Properties is a hash describing the templates
+        # @see https://cloud.google.com/compute/docs/reference/rest/v1/instanceTemplates/insert
+        attribute :properties
+
+        def save
+          requires :name
+          requires :properties
+
+          data = service.insert_instance_template(name, properties, description)
+          operation = Fog::Compute::Google::Operations.new(:service => service).get(data.name)
+          operation.wait_for { !pending? }
+          reload
+        end
+
+        def destroy(async = true)
+          requires :name
+          operation = service.delete_instance_template(name)
```

Comment:
You need to return a Fog::Compute::Google::Operation model here, see your `save` method.
Otherwise you'll get a Google API client class that doesn't respond to `.wait_for`:
```
NoMethodError: undefined method `wait_for' for #<Google::Apis::ComputeV1::Operation:0x00007fd00aed3f00>
```

Reply:
Done